pkgs/development/python-modules/pysaml2/default.nix

```diff
@@ -21,7 +22,16 @@
 , xmlsec
 }:
 
-buildPythonPackage rec {
+let
+  pymongo3 = pymongo.overridePythonAttrs(old: rec {
+    version = "3.12.3";
+    src = fetchPypi {
+      pname = "pymongo";
+      inherit version;
+      sha256 = "sha256-ConK3ABipeU2ZN3gQ/bAlxcrjBxfAJRJAJUoL/mZWl8=";
+    };
+  });
```

Comment:
We don't override packages inside python-modules, because it is easily results in conflicts when multiple packages depend on different versions while only one can claim the name inside the python environment.

If you want to fix matrix-synapse OTOH, an override there would be acceptable.

Reply:
This is only used in tests (and in checkInputs), so I think it's fine here?

Reply:
Ah well, if it's not propagated that should be fine.

Reply:
It is not, and it doesn't look like anything in nixpkgs uses both pysaml2 and pymongo at the same time.

Reply:
> it doesn't look like anything in nixpkgs uses both pysaml2 and pymongo at the same time.

but something could use it in the future. Don't we have a hook now that detects it? But I don't think it is perfect.

As long as this never gets propagated it should be fine :tm:

Reply:
Something can even get away with using both, as long as it doesn't specifically use pymongo _and_ the mongo storage for pysaml2, but if you're doing that, you'll explode anyway.